crates/noirc_frontend/src/node_interner.rs

```diff
@@ -355,9 +348,9 @@ impl NodeInterner {
         f(&mut value);
     }
 
-    pub fn update_type_alias(&mut self, type_id: TyAliasId, f: impl FnOnce(&mut TypeAliasTy)) {
-        let mut value = self.type_aliases.get_mut(&type_id).unwrap().borrow_mut();
-        f(&mut value);
+    pub fn update_type_alias(&mut self, type_id: TypeAliasId, f: impl FnOnce(&mut TypeAliasType)) {
+        let value = self.type_aliases.get_mut(&type_id).unwrap();
+        f(value);
```

Comment:
I don't think this function will be needed at all after the ID change. Maybe just a single `set_type_alias_type` method.